File: src/main/java/dev/boooiil/historia/core/runnable/UpdateScoreboardRunnable.java
package dev.boooiil.historia.core.runnable;

import dev.boooiil.historia.core.database.internal.PlayerStorage;
import dev.boooiil.historia.core.player.HistoriaPlayer;
import dev.boooiil.historia.core.scoreboard.ScoreboardAdapter;
import dev.boooiil.historia.core.util.NumberUtils;
import net.md_5.bungee.api.ChatColor;
import org.bukkit.Bukkit;
import org.bukkit.scheduler.BukkitRunnable;

public class UpdateScoreboardRunnable extends BukkitRunnable {

        @Override
        public void run() {
                Bukkit.getOnlinePlayers().forEach(player -> {

                        HistoriaPlayer historiaPlayer = PlayerStorage.getPlayer(player.getUniqueId());
                        ScoreboardAdapter scoreboardAdapter = new ScoreboardAdapter();

                        scoreboardAdapter.createHeader(ChatColor.GOLD + "" + ChatColor.BOLD + "HISTORIA");
                        scoreboardAdapter.addLine(12, "");
                        scoreboardAdapter.addLine(11,
                                        ChatColor.AQUA + "Proficiency: " + ChatColor.GRAY
                                                        + historiaPlayer.getProficiency().getName().getKey());
                        scoreboardAdapter.addLine(10,
                                        ChatColor.DARK_AQUA + "Level: " + ChatColor.GRAY + historiaPlayer.getLevel());
                        scoreboardAdapter.addLine(9, ChatColor.AQUA + "Health: " + ChatColor.GRAY
                                        + NumberUtils.roundDouble(player.getHealth(), 2) + "/"
                                        + historiaPlayer.getBaseHealth());
                        scoreboardAdapter.addLine(8,
                                        ChatColor.DARK_AQUA + "Hunger: " + ChatColor.GRAY
                                                        + NumberUtils.roundDouble(player.getFoodLevel(), 2) + "/"
                                                        + historiaPlayer.getProficiency().getStats().getBaseFood());
                        scoreboardAdapter.addLine(7, ChatColor.AQUA + "Experience: " + ChatColor.GRAY
                                        + historiaPlayer.getCurrentExperience() + "/"
                                        + historiaPlayer.getMaxExperience());
                        scoreboardAdapter.addLine(6,
                                        ChatColor.DARK_AQUA + "Temperature: " + ChatColor.GRAY
                                                        + historiaPlayer.getCurrentTemperature());
                        scoreboardAdapter.addLine(5, ChatColor.AQUA + "Weight: " + ChatColor.GRAY + null);
                        scoreboardAdapter.addLine(4, ChatColor.DARK_AQUA + "Weapon Class: " + ChatColor.GRAY
                                        + historiaPlayer.getProficiency().getStats().getUsableWeaponTypes());
                        scoreboardAdapter.addLine(3, ChatColor.AQUA + "Armor Class: " + ChatColor.GRAY
                                        + historiaPlayer.getProficiency().getStats().getUsableArmorTypes());

                        scoreboardAdapter.addToPlayer(player);

                });
        }

}

File: src/main/java/dev/boooiil/historia/core/runnable/UpdateScoreboardRunnable.kt
package dev.boooiil.historia.core.runnable

import dev.boooiil.historia.core.database.internal.PlayerStorage
import dev.boooiil.historia.core.scoreboard.ScoreboardBuilder
import dev.boooiil.historia.core.util.NumberUtils
import dev.boooiil.historia.core.util.plus
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor
import net.kyori.adventure.text.format.TextDecoration
import org.bukkit.Bukkit
import org.bukkit.entity.Player
import org.bukkit.scheduler.BukkitRunnable

class UpdateScoreboardRunnable : BukkitRunnable() {

    override fun run() {
        Bukkit.getOnlinePlayers().forEach { player: Player ->
            val historiaPlayer = PlayerStorage.getPlayer(player.uniqueId)

            player.scoreboard = ScoreboardBuilder()
                .header(Component.text("HISTORIA", NamedTextColor.GOLD).decorate(TextDecoration.BOLD))

                .addLine(Component.empty())

                .addLine(Component.text("Proficiency: ", NamedTextColor.AQUA)
                        + historiaPlayer.getProficiency().displayName.color(NamedTextColor.GRAY))

                .addLine(Component.text("Level: ", NamedTextColor.DARK_AQUA)
                        + Component.text(historiaPlayer.level, NamedTextColor.GRAY))

                .addLine(Component.text("Health: ", NamedTextColor.AQUA)
                        + Component.text("${NumberUtils.roundDouble(player.health, 2)}/${historiaPlayer.baseHealth}", NamedTextColor.GRAY))

                .addLine(Component.text("Hunger: ", NamedTextColor.DARK_AQUA)
                        + Component.text("${NumberUtils.roundDouble(player.foodLevel.toDouble(), 2)}/0", NamedTextColor.GRAY))

                .addLine(Component.text("Experience: ", NamedTextColor.AQUA)
                        + Component.text("${historiaPlayer.currentExperience}/${historiaPlayer.maxExperience}", NamedTextColor.GRAY))

                .addLine(Component.text("Temperature: ", NamedTextColor.DARK_AQUA)
                        + Component.text(historiaPlayer.currentTemperature.toString(), NamedTextColor.GRAY))

                .addLine(Component.text("Weight: ", NamedTextColor.AQUA)
                        + Component.text("0", NamedTextColor.GRAY)) // TODO placeholder weight value

                .build()

            // scoreboardAdapter.addLine(4, ChatColor.DARK_AQUA + "Weapon Class: " +
            // ChatColor.GRAY
            // + historiaPlayer.getProficiency().getStats().getWeaponStats()
            // .getUsableWeaponWeights());
            // scoreboardAdapter.addLine(3, ChatColor.AQUA + "Armor Class: " +
            // ChatColor.GRAY
            // + historiaPlayer.getProficiency().getStats().getArmorStats()
            // .getUsableArmorWeights());
        }
    }
}
